packages/react-components/src/lib/hooks/amountInput.ts

```diff
@@ -0,0 +1,49 @@
+import type { NatValue } from '@agoric/ertp/src/types';
+import { AssetKind } from '@agoric/ertp';
+import { parseAsValue, stringifyValue } from '@agoric/web-components';
+import { useState } from 'react';
+
+type Args = {
+  value: NatValue | null;
+  decimalPlaces: number;
+  onChange: (value: NatValue) => void;
+};
+
+export const useAmountInput = ({ value, decimalPlaces, onChange }: Args) => {
+  const amountString = stringifyValue(value, AssetKind.NAT, decimalPlaces);
+
+  const [fieldString, setFieldString] = useState(
+    value === null ? '' : amountString,
+  );
+
+  const handleInputChange: React.ChangeEventHandler<HTMLInputElement> = ev => {
```

Comment:
A future improvement might be adding debounce, as this is maybe expensive on every keystroke, but LGTM.

Reply:
Hmm I wonder how that would work with the displayString updating, but yea it's not too much of a perf issue

Reply:
Yes brain fart this does not make sense in that context